code/citadel/cit_arousal.dm

```diff
@@ -181,224 +181,365 @@
+							"<span class='userdanger'>Your [G] cannot cum, giving no relief.</span>")
+	else
+		total_fluids = fluid_source.total_volume
+		src.visible_message("<span class='danger'>[src] looks like they're about to cum.</span>", \
+							"<span class='green'>You feel yourself about to orgasm.</span>", \
+							"<span class='green'>You feel yourself about to orgasm.</span>")
+		if(do_after(src, mb_time, target = src))
+			if(total_fluids > 5)
+				fluid_source.reaction(src.loc, TOUCH, 1, 0)
+			fluid_source.clear_reagents()
+			src.visible_message("<span class='danger'>[src] orgasms[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""], using [p_their()] [G]!</span>", \
+								"<span class='green'>You climax[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""] with your [G].</span>", \
+								"<span class='green'>You climax using your [G].</span>")
+			if(G.can_climax)
+				setArousalLoss(min_arousal)
+
+
+/mob/living/carbon/human/proc/mob_climax_partner(obj/item/organ/genital/G, mob/living/L, spillage = TRUE) //Used for climaxing with any living thing
+	var/mb_time = 30
+	var/total_fluids = 0
+	var/datum/reagents/fluid_source = null
+
+	if(G.producing) //Can it produce its own fluids, such as breasts?
+		fluid_source = G.reagents
+	else
+		if(!G.linked_organ)
+			to_chat(src, "<span class='warning'>Your [G] is unable to produce it's own fluids, it's missing the organs for it.</span>")
+			return
+		fluid_source = G.linked_organ.reagents
+	total_fluids = fluid_source.total_volume
+
+	src.visible_message("[src] is about to climax with [L]!", \
+						"You're about to climax with [L]!", \
+						"<span class='danger'>You're preparing to climax with someone!</span>")
+	if(spillage)
+		if(do_after(src, mb_time, target = src) && in_range(src, L))
+			fluid_source.trans_to(L, total_fluids*G.fluid_transfer_factor)
+			total_fluids -= total_fluids*G.fluid_transfer_factor
+			if(total_fluids > 5)
+				fluid_source.reaction(L.loc, TOUCH, 1, 0)
+			fluid_source.clear_reagents()
+			src.visible_message("<span class='danger'>[src] climaxes with [L][spillage ? ", overflowing and spilling":""], using [p_their()] [G]!</span>", \
+								"<span class='green'>You orgasm with [L][spillage ? ", spilling out of them":""], using your [G].</span>", \
+								"<span class='green'>You have climaxed with someone[spillage ? ", spilling out of them":""], using your [G].</span>")
+	else //knots and other non-spilling orgasms
+		if(do_after(src, mb_time, target = src) && in_range(src, L))
+			fluid_source.trans_to(L, total_fluids)
+			total_fluids = 0
+			src.visible_message("<span class='danger'>[src] climaxes with [L], [p_their()] [G] spilling nothing!</span>", \
+								"<span class='green'>You ejaculate with [L], your [G] spilling nothing.</span>", \
+								"<span class='green'>You have climaxed inside someone, your [G] spilling nothing.</span>")
+	if(G.can_climax)
```

Comment:
Fixed. I added the check to the spillage also. I think my idea was for the command to run after either the spillage and non-spillage versions, but I forgot about the do_after part being important for both.